libethcore/Precompiled.cpp

```diff
@@ -132,15 +132,34 @@ ETH_REGISTER_PRECOMPILED(modexp)(bytesConstRef _in)
 	return {true, ret};
 }
 
+namespace
+{
+	bigint expLengthAdjust(bigint _baseLength, bigint _expLength, bytesConstRef _in)
+	{
+		size_t const numConsideredBytes(min(_expLength, {32}));
+		bigint expConsideredBytes{};
+		if (96 + _baseLength == bigint(size_t(96 + _baseLength)))
```

Comment:
Then I guess we need to add checks that lengths fit into `size_t` into modexp implementation itself.

Also I think I would use `std::numeric_limits` for that instead of C-style-cast tricks